src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -446,7 +451,7 @@ private BoundExpression CheckValue(BoundExpression expr, BindValueKind valueKind
                 var resolution = this.ResolveMethodGroup(methodGroup, analyzedArguments: null, isMethodGroupConversion: false, useSiteInfo: ref useSiteInfo);
                 diagnostics.Add(expr.Syntax, useSiteInfo);
                 Symbol otherSymbol = null;
-                bool resolvedToMethodGroup = resolution.MethodGroup != null;
+                bool resolvedToUnusableSymbol = resolution.MethodGroup == null && resolution.ResultKind != LookupResultKind.Viable;
```

Comment:
> && resolution.ResultKind != LookupResultKind.Viable [](http://example.com/codeflow?start=78&length=52)

It is not obvious why this change is necessary.

Reply:
With the previous logic, in an extension member scenario (ie. viable result, but no method group), we'd end up returning a `BoundBadExpression` at line 480 below.
We may have no method group, but we have an extension member, so should not go into error handling branch.

Would it be clearer with the equivalent `bool resolvedToUnusableSymbol = resolution.MethodGroup == null && !resolution.IsExtensionMember(out _);`?

Reply:
> Would it be clearer with the equivalent ```bool resolvedToUnusableSymbol = resolution.MethodGroup == null && !resolution.IsExtensionMember(out _);```?

Probably. Or a comment might be helpful